scalyr_agent/configuration.py

```diff
@@ -3905,6 +3905,15 @@ def __verify_monitor_entry_and_set_defaults(
             monitor_entry, "log_path", module_name + ".log", description
         )
 
+        # Verify option which tells that agent has to fail if that monitor stops.
+        self.__verify_or_set_optional_bool(
+            monitor_entry,
+            field="stop_agent_if_fails",
```

Comment:
Perhaps ``stop_agent_on_failure`` is a better name

Reply:
I assume we will define k8s monitor as essential as part of the Kubernetes deployment config, right?

Also, I wonder if simply ``essential`` may be a better config option name (although, on the other hand, I kinda prefer explicit nature of the ``stop_agent_on_failure`` option name).

Reply:
> I assume we will define k8s monitor as essential as part of the Kubernetes deployment config, right?
correct,  I just have next coming PR where I extend pytest ent-to-end tests with docker end k8s tests, and I want to include that there.

> Also, I wonder if simply essential may be a better config option name (although, on the other hand, I kinda prefer explicit nature of the stop_agent_on_failure option name).

yes, I am also on the fence here, and I also thought that explicit naming would be easier for customers understanding.